DiamondLightSource/hyperion
# 1073 make external callbacks optional

Fixes #1073

Requires GDA change https://gerrit.diamond.ac.uk/c/gda/gda-mx/+/40900 to take effect, but shouldn't break anything on its own

### To test:
1. Run tests
2. Run `hyperion_restart()` from GDA (with the above change applied) and see that hyperion is started with the external callbacks turned off
